docs/standard/library-guidance/dependencies-net.md

```diff
@@ -0,0 +1,152 @@
+
+**Pros**
+- Simplifies decision-making and aligns with stability expectations.
+- Reduces complexity in dependency management.
+- Encourages modernization and access to new features.
+
+**Cons**
+- Older TFMs may require app-local deployments, increasing app size.
+- Larger number of packages to update in comparison to relying on framework provided API.
+- May create friction for customers who are not on the latest supported runtime.
+- May prevent consumption in enviroments where dependencies are managed by a host application - eg: MSBuild, Visual Studio, Azure Functions V1.
+
+---
+
+### **Option 2: TFM-Specific Versions**
+Reference different dependency versions per Target Framework Moniker (TFM).  Don't reference packages when the framework provides the API.  For example: 8.0 packages on `net8.0`, 9.0 packages on `net9.0`, and so on.
+
+**Pros**
+- Minimizes change for apps running on older runtimes.
+- Minimizes app-local libraries app size on older runtimes.  Uses runtime optimized library on older runtimes.
+
+**Cons**
+- Reduced API available to library which may lead to more complex implementations (polyfills).  Polyfills increase the total cost of engineering and servicing.
+- Slows innovation in libraries.
+- Greater complexity of infrastructure to maintain seperate dependency sets per target framework.  Central packagge management and dependabot can be be configured to work with this, but it's challenging to get it right.
+
+---
+
+### **Option 3: Branching**
+Reference the latest supported version of the dependency across all target frameworks, but branch your library in sync with target frameworks.  This is the same as Option 1, but each time a new framework is added, a new major version / branch is created to allow for updating the major version of dependencies and adding the new target framework.
+
+**Pros**
+- Balances compatibility with flexibility for fast-moving areas.
+- Encourages modernization and access to new features.
+- Simplifies decision-making and aligns with stability expectations.
+- Reduces complexity in dependency management.
+
+**Cons**
+- Increased infrastructure cost to maintain concurrent branches.
+- No new features for folks who want to stay on older dependencies.
+
+---
+
+## **Decision Matrix**
+
+| Strategy                              | Update Friction  | Engineering Cost |  Servicing Cost  |
+|---------------------------------------|------------------|------------------|------------------|
+| Latest Supported Versions (Option 1)  | Moderate         | Low              | Moderate         |
+| TFM-Specific Versions (Option 2)      | Low              | High             | Low / Moderate   |
+| Branching                             | Low              | Low              | High             |
```

Comment:
```suggestion
| Branching                             | Low              | High             | High             |
```

Reply:
No, the engineering cost of the branching solution is low.  Here I use "engineering" to mean steady-state engineering cost in main and I contrast that with "servicing cost".

Reply:
I'm wondering if we should even give this option. What is an example of a library that would follow this? Where they basically only add new functionality to the latest TFM support? Are there any examples outside of the .NET Team?

Reply:
This is the option for any product that leverages branches to control risk. I do think it's important to call this out, since it does relate to plenty of folks asking these questions, and shows customers how we solve this problem.  I can run a scan of nuget to see which packages use branching like this.

Reply:
In the top 100 non-microsoft packages, the following use branches to manage risk, determined by observing lower major versions published after higher major versions.
```
AWSSDK.Core
Castle.Core
Moq
AutoMapper
FluentValidation
Npgsql
FluentAssertions
NUnit
NLog
AWSSDK.S3
RabbitMQ.Client
FluentValidation.DependencyInjectionExtensions
Npgsql.EntityFrameworkCore.PostgreSQL
Elasticsearch.Net
AWSSDK.SecurityToken
FluentValidation.AspNetCore
NodaTime
jQuery
MessagePack
AWSSDK.SQS
```

Reply:
> Maybe I can add a note under branching

That sounds good to me. Thanks